File: csharp/UrakawaToolkit/ITextMedia.cs
using System;

namespace urakawa.media
{
	/// <summary>
	/// Text media conforms to this interface.
	/// </summary>
	public interface ITextMedia : IMedia
	{
		/// <summary>
		/// Get the text string for the TextMedia.
		/// </summary>
		/// <returns></returns>
		string getText();

		/// <summary>
		/// Set the text string for the TextMedia.
		/// Throws <see cref="urakawa.exception.MethodParameterIsNullException"/>, 
		/// <see cref="urakawa.exception.MethodParameterIsEmptyStringException"/>
		/// </summary>
		/// <param name="text">The new text string.</param>
		void setText(string text);
	}
}

File: java/core/src/org/daisy/urakawa/media/ITextMedia.java
package org.daisy.urakawa.media;

import org.daisy.urakawa.exception.MethodParameterIsNullException;

/**
 * The simple text media type, which stores text data internally (no external
 * location)
 * 
 * @leafInterface see {@link org.daisy.urakawa.LeafInterface}
 * @see org.daisy.urakawa.LeafInterface
 * @stereotype OptionalLeafInterface
 */
public interface ITextMedia extends IMedia
{
    /**
     * @return the text. Cannot be NULL
     */
    public String getText();

    /**
     * @param text Cannot be NULL
     * @throws MethodParameterIsNullException NULL method parameters are
     *         forbidden
     * 
     * @tagvalue Events "TextChanged"
     */
    public void setText(String text) throws MethodParameterIsNullException;
}
